include/envoy/tcp/BUILD

```diff
@@ -17,7 +17,7 @@ envoy_cc_library(
 
 envoy_cc_library(
     name = "conn_pool_interface",
-    hdrs = ["conn_pool.h"],
+    hdrs = ["conn_pool_nats.h"],
```

Comment:
Consider adding a TODO explaining why this is necessary for now and that it needs to be changed once `conn_pool.h` in envoyproxy/proxy is ready.